docs/agent.md

```diff
@@ -4,43 +4,100 @@ The agent is responsible for the life cycle of the computation, i.e., running th
 
 The picture below shows where the agent runs in the Cocos system, helping us better understand its role.
 
-![Agent](/img/agent.png)
+![Agent](/img/agent/overview.png)
 
 ## StateMachine
 
-- Orchestrates the overall flow of the computation.
-- Transitions between states based on received events.
-- Defines valid state transitions and associated functions.
+The agent operates like a state machine.
+This state machine defines valid states, transitions between these states based on received events and functions associated with these transitions.
+
+The picture below show the overall flow of the computation.
+
+![Agent](/img/agent/state_machine.png)
 
 ### States
 
-- `idle`: Initial state, waiting for the computation to start.
-- `receivingManifest`: Receives the initial computation manifest.
-- `receivingAlgorithm`: Receives the algorithm for the computation.
-- `receivingData`: Receives dataset data for the computation.
-- `running`: Executes the computation using received algorithms and data.
-- `resultsReady`: Computation has finished, results are available.
-- `complete`: All results have been consumed, computation lifecycle ends.
+- `Idle`: Initial state, waiting for the computation to start.
+- `ReceivingManifest`: Receives the initial computation manifest. The computation manifest contains all the information necessary for the computaion run. Most important pieces of information are the hash of the algorithm, number of datasets and the hash of each dataset.
+- `ReceivingAlgorithm`: Receives the algorithm for the computation.
+- `ReceivingData`: Receives dataset data for the computation. The agent remains in this state until it receives all the datasets. The number of datasets is determined by the computaion manifest. Upon each upload, the hash of the incoming dataset is compared to the hash of the dataset sent in the manifest.
+- `Running`: Executes the computation using received algorithms and data.
+- `ConsumingResults`: Computation has finished, results are available.
+- `Complete`: All results have been consumed, computation lifecycle ends.
+- `Failed`: An error was encountered during the computation run.
 
 ### Events
 
-- `start`: Triggers the computation startup process.
-- `manifestReceived`: Indicates computation manifest has been received.
-- `algorithmReceived`: Indicates the algorithm has been received.
-- `dataReceived`: Indicates all dataset data has been received.
-- `runComplete`: Signals the completion of the computation execution.
-- `resultsConsumed`: Indicates all consumers have retrieved the results.
+- `Start`: Triggers the computation startup process.
+- `ManifestReceived`: Indicates computation manifest has been received.
+- `AlgorithmReceived`: Indicates the algorithm has been received. During the upload the hash of the incoming algorithm is compared to the hash of the algorithm sent in the manifest. If datasets are expected the Agent moves into `ReceivingData` state, otherwise it moves in `Running` state.
```

Comment:
This sentence talks about receiving datasets and algorithms; it is confusing. Keep them on separate points. one about receiving algo and comparing checksum/hash and another for datasets